### ISSUE TITLE
[Bug] Send email node sends duplicate emails

### ISSUE THREAD

**Issue by OzyOzk** [thingsboard/thingsboard#7207]
**Describe the bug**
The send email node in the rule chain sends the same email twice. Issue is inconsistent and occurs in roughly 1 in 2 emails.

**Your Server Environment**

* own setup
  * AWS EC2
  * 3.3.4.1PE-1
  * AWS Linux

**Your Client Environment**

**Desktop (please complete the following information):**

* OS: Windows 10
* Browser FireFox
* Version 104.0.1

**To Reproduce**

I'm unable to reproduce this behavior. When I use virtual devices to trigger alarms, the emails are never duplicated.

**Expected behavior**
Rule-chain should only be sending one email per alarm.

**Screenshots**
![rulechain error](https://user-images.githubusercontent.com/40250200/188466382-7c77e9d4-74f5-46b3-8718-0082b4177a4f.PNG)
 explain your problem.

**Context**

This rule-chain is designed to only raise an alarm if the co2 value has been above a threshold for at least 15 minutes.

**Comment by Backdraft007** [comment]
I would recommend to enable debug modus in switch node and create alarm node to check whether the device sends data twice.

**Comment by OzyOzk** [comment]
> I would recommend to enable debug modus in switch node and create alarm node to check whether the device sends data twice.

I already checked if we're getting duplicate messages. The issue seems to be caused by the label going from the create alarm node to the script node. I had these labels as "created/updated". I changed them to "created" only and now I'm not getting duplicate emails, though I can't say for certain that this was the cause.

**Comment by Backdraft007** [comment]
I think it is the cause. I cannot test it in the moment.

**Comment by OzyOzk** [comment]
> I think it is the cause. I cannot test it in the moment.

This was not the cause unfortunately. I'm still occasionally getting double emails. I turned on debug and saw that there were two identical messages going into the send email node. Perhaps this is caused by the save initial alarm time node on the bottom left. I've since moved that so it comes after the create alarm node. Hopefully this solves it.

**Comment by netbc** [comment]
We are noticing similar issue, the problem might be the switch node when TBEL is used:
![image](https://user-images.githubusercontent.com/110825868/226616450-45fc0565-7102-4177-8df7-bd5383379b30.png)
![image](https://user-images.githubusercontent.com/110825868/226617702-ef07cabc-6949-4ae4-bcca-5dc5f23be07b.png)
![image](https://user-images.githubusercontent.com/110825868/226618395-60845140-68af-49a9-9652-418477cdd18a.png)
According to the log two messages gets out from switch node but five messages are received in the next node. Relation type is missing in incoming messages.

**Comment by SimoneG97** [comment]
I know this is an old issue, but I think I've found myself in the same situation but without emails, just duplicate messages.
Keep in mind this happened only once until now, but I can't find myself explain this behaviour.
I'll only describe the part where I've found this issue because the rest is kinda irrelevant, but if you need it I will provide.
![immagine](https://github.com/thingsboard/thingsboard/assets/38068023/9f9c61d1-d55c-4f7b-973d-7c73e8c84ea2)
One particular message arrived at the switch block, its content is also irrelevant.
![immagine](https://github.com/thingsboard/thingsboard/assets/38068023/4b937703-7230-434b-b07f-89bdf611123d)
One in, one out. So far everything is behaving normally.
But something weird happens when the message is "duplicated" to the 2 script blocks.
![immagine](https://github.com/thingsboard/thingsboard/assets/38068023/e3fae441-bd8a-4a20-9501-9975dd71e5b1)
![immagine](https://github.com/thingsboard/thingsboard/assets/38068023/de8e9878-b886-42d9-8d46-43d4c806c49d)
This is where the anomaly happens, both nodes received 2 identical messages, not just 1, effectively quadrupling the original message.
I have not been able to reproduce this behaviour, it seems to happen randomly.
The reason I have reported it here is because the structure of the rule chain is strikingly similar, as in having a node which duplicates messages.

The version of Thingsboard (with Cassandra) used is 3.6.1, installed on an Ubuntu machine with Docker.

**Comment by mykhailokornieiev** [comment]
As the community has not provided assistance on this matter over an extended period, I am closing this ticket due to inactivity. If it is still relevant, feel free to open a new ticket with a full description of your request.
